File: 0013-roman-to-integer/13.roman-to-integer.cs
/*
 * @lc app=leetcode id=13 lang=csharp
 *
 * [13] Roman to Integer
 */

// @lc code=start
public class Solution {
    public int RomanToInt(string s) {
        var romDict = new Dictionary<char, int>() {
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}
        };

        int total = 0;
        for (int i = 0; i < s.Length; i++)
        {
            if (i != s.Length - 1 && romDict[s[i+1]] > romDict[s[i]])
            {
                total += romDict[s[i+1]] - romDict[s[i]];
                i++;
            }
            else
            {
                total += romDict[s[i]];
            }
        }

        return total;
    }
}
// @lc code=end

File: 0013-roman-to-integer/13.roman-to-integer.py
#
# @lc app=leetcode id=13 lang=python3
#
# [13] Roman to Integer
#

# @lc code=start
class Solution:
    def romanToInt(self, s: str) -> int:
        rom_dict = {
            "I" : 1,
            "V" : 5,
            "X" : 10,
            "L" : 50,
            "C" : 100,
            "D" : 500,
            "M" : 1000
        }

        total = 0
        i = 0
        while i < len(s):
            if i != len(s) - 1 and rom_dict[s[i+1]] > rom_dict[s[i]]:
                total += rom_dict[s[i+1]] - rom_dict[s[i]]
                i += 2
            else:
                total += rom_dict[s[i]]
                i += 1

        return total

# @lc code=end

